pkg/core/dao/dao.go

```diff
@@ -616,9 +615,31 @@ func (dao *Simple) StoreAsCurrentBlock(block *block.Block, buf *io.BufBinWriter)
 	return dao.Store.Put(storage.SYSCurrentBlock.Bytes(), buf.Bytes())
 }
 
-// StoreAsTransaction stores given TX as DataTransaction. It can reuse given
+// StoreAsTransaction stores given TX as DataTransaction. It also stores transactions
+// given tx has conflicts with as DataTransaction with dummy version. It can reuse given
 // buffer for the purpose of value serialization.
-func (dao *Simple) StoreAsTransaction(tx *transaction.Transaction, index uint32, buf *io.BufBinWriter) error {
+func (dao *Simple) StoreAsTransaction(tx *transaction.Transaction, index uint32, p2pSigExtensions bool, buf *io.BufBinWriter) error {
+	err := dao.storeTransaction(tx, index, buf)
+	if err != nil {
+		return err
+	}
+	if p2pSigExtensions {
+		for _, attr := range tx.GetAttributes(transaction.ConflictsT) {
+			buf.Reset()
+			hash := attr.Value.(*transaction.Conflicts).Hash
+			dummyTx := transaction.NewTrimmedTX(hash)
```

Comment:
So we allocate a struct only to encode 1 constant byte after? Key may also be reused.

Reply:
Fixed, benchmark results for these approaches are in https://github.com/nspcc-dev/neo-go/pull/2134#issuecomment-900975777.